src/assignments/benson/lesson_2_answers.clj

```diff
@@ -1,3 +1,59 @@
 (ns lesson-2-answers)
 
-;;; Enter your answers in here.
+;; Implements filter using a loop
+(defn filter-loop
+  [pred input]
+  ;; recurses through input sequence and conjoins all variables
+  ;; that pass the predicate condition
+  (loop [xs input
+         result []]
+    (if xs
+      (let [x (first xs)]
+        (if (pred x)
+          (recur (next xs) (conj result x))
+          (recur (next xs) result)))
+      result)))
+
+
+;; Implements filter using reduce
+(defn filter-reduce
+  [pred input]
+  (reduce
+   (fn [output x]
+     ;; Conjoin all values that pass the predicate condition
+     (if (pred x)
+       (conj output x)
+       output))
+   []
+   input))
+
+
+;; Implements update-in function
+(defn update-in-custom
+  [m k f]
+  ;; uses assoc-in but applies inputted function
+  (assoc-in
+   m k
+   (f (get-in m k))))
+
+
+;; Finds and returns sets of anagrams encapsulated in a set
+(defn find-anagram
```

Comment:
Very elegant!

The only criticism I would have might be the name: find-anagram ?
Maybe something like group-by-anagram might by better?  In any
case, well done!  Probably nicer that the solution I will present in class,
in some ways!  (and awesomely better than what you showed on Thursday, no?) ;-)